src/Execution/ExecutionStrategy.php

```diff
@@ -238,9 +241,23 @@ protected function executeFields(
                 continue;
             }
 
+            if (!$isContainsPromise && $this->isPromise($result)) {
+                $isContainsPromise = true;
+            }
+
             $finalResults[$fieldName] = $result;
         }
 
+        if ($isContainsPromise) {
+            $keys    = array_keys($finalResults);
+            $promise = \React\Promise\all(array_values($finalResults));
+            $promise->then(function ($values) use ($keys, &$finalResults) {
```

Comment:
@crisu83 what do you think about passing `$finalResults` as ref?

Reply:
I think it's acceptable in this case, however, it might be a little bit slower. I would use `array_reduce` instead.